src/test/coloridentifier_tests.cpp

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2020 Chaintope Inc.
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+
+#include <test/test_tapyrus.h>
+#include <map>
+#include <primitives/transaction.h>
+#include <script/script.h>
+#include <coloridentifier.h>
+#include <boost/test/unit_test.hpp>
+
+BOOST_FIXTURE_TEST_SUITE(coloridentifier_tests, BasicTestingSetup)
+
+BOOST_AUTO_TEST_CASE(coloridentifier_valid_unserialize)
+{
+    //type NONE
+    ColorIdentifier c0;
+    uint8_t str[32] = {};
+    CDataStream ss0(ParseHex("00"), SER_NETWORK, INIT_PROTO_VERSION);
+    ss0 >> c0;
+    BOOST_CHECK_EQUAL(TokenToUint(c0.type), TokenToUint(TokenTypes::NONE));
+    BOOST_CHECK(memcmp(&c0.payload[0], &str[0], 32) == 0);
+
+    try {
+        CDataStream ss00(ParseHex("0100"), SER_NETWORK, INIT_PROTO_VERSION);
+        ss00 >> c0;
+        BOOST_CHECK_MESSAGE(false, "We should have thrown");
+    } catch (const std::ios_base::failure& e) {
+    }
+
```

Comment:
Can you add a test for 33 bytes of data starting with an invalid color type(0x04)?